repositories/twitch.go

```diff
@@ -0,0 +1,93 @@
+package repositories
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/streampets/backend/models"
+)
+
+type usernameResponse struct {
+	Data []struct {
+		Login string `json:"login"`
+	} `json:"data"`
+}
+
+type accessTokenResponse struct {
+	AccessToken string `json:"access_token"`
+}
+
+type TwitchRepository struct {
+	ClientID     string
+	ClientSecret string
+}
+
+func NewTwitchRepository(id, secret string) *TwitchRepository {
+	return &TwitchRepository{ClientID: id, ClientSecret: secret}
+}
+
+func (repo *TwitchRepository) GetUsername(userID models.UserID) (string, error) {
+	url := fmt.Sprintf("https://api.twitch.tv/helix/users?id=%s", userID)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", err
+	}
+
+	accessToken, err := repo.getAccessToken()
```

Comment:
The app access token should probably be retrieved on creation or within a separate call after creation.  Presumably you're going to have more methods than `GetUsername` and the app token stays valid for a longer period of time than user tokens.   Don't think you want to re-call it each time someone calls `GetUsername`.  Not sure what the `Go` way of doing this would be, but normally you would either setup a timer based on the expiry date, or automatically refresh on the first 401 and try again.